src/main/java/com/faforever/client/coop/CoopController.java

```diff
@@ -194,6 +200,20 @@ public void initialize() {
     });
   }
 
+  private void onSearchFieldChange(String newvalue){
+    leaderboardFilteredTable.setPredicate(coopResultBean -> {
+      String teamstring = "";
+      for(Map.Entry<String, List<String>> entry : coopResultBean.getReplay().getTeams().entrySet()){
+        teamstring = entry.getValue().stream().collect(Collectors.joining(i18n.get("textSeparator")));
+      }
+      if(newvalue.isEmpty() || teamstring.equalsIgnoreCase("")){
+        return true;
+      }
+      return teamstring.toLowerCase().contains(newvalue.toLowerCase());
+    });
```

Comment:
We are looking for a team member in the list of teams. So it would be more efficient to use the anyMatch of a stream rather than concatenating all the names together and doing a contains.

Reply:
Additionally we can set the predicate in the initialize function by binding it to the text field value.

Reply:
Also it looks like this only checks the last team in the coop result bean instead of all teams